### PR TITLE
windsurf 1.0.5 (new cask)

### PR DESCRIPTION
**Pull request by fathiraz**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/w/windsurf.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  url "https://windsurf-stable.codeiumdata.com/#{arch}-dmg/stable/#{version.csv.second}/Windsurf-#{arch}-#{version.csv.first}.dmg",
+      verified: "windsurf-stable.codeiumdata.com/"
+  name "Windsurf"
+  desc "Agentic IDE powered by AI Flow paradigm"
+  homepage "https://www.codeium.com/windsurf"
+
+  livecheck do
+    url "https://windsurf-stable.codeium.com/api/update/#{arch}-dmg/stable/latest"
+    strategy :json do |json|
+      "#{json["windsurfVersion"]},#{json["version"]}"
```
Comment (khipp) [inline Casks/w/windsurf.rb:17]:
```suggestion
      version = json["windsurfVersion"]
      hash = json["version"]
      next if version.blank? || hash.blank?

      "#{version},#{hash}"
```

File: Casks/w/windsurf.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+cask "windsurf" do
+  arch arm: "darwin-arm64", intel: "darwin-x64"
```
Comment (khipp) [inline Casks/w/windsurf.rb:2]:
```suggestion
  arch arm: "arm64", intel: "x64"
```

This change allows for more flexibility in the future, as many apps also use short identifiers, e.g., in API responses.

File: Casks/w/windsurf.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  version "1.0.5,d33d40f6cd3a4d7e451b22e94359230a4aa8c161"
+  sha256 arm:   "706541a254753cb7ba8d371fa39753ca7075cda2917907da46fe535bb209e304",
+         intel: "622d68b0df1427e41211031db53b2231976359d5099c7346e4028f72a95e202b"
+
+  url "https://windsurf-stable.codeiumdata.com/#{arch}-dmg/stable/#{version.csv.second}/Windsurf-#{arch}-#{version.csv.first}.dmg",
```
Comment (khipp) [inline Casks/w/windsurf.rb:8]:
```suggestion
  url "https://windsurf-stable.codeiumdata.com/darwin-#{arch}-dmg/stable/#{version.csv.second}/Windsurf-darwin-#{arch}-#{version.csv.first}.dmg",
```

File: Casks/w/windsurf.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      verified: "windsurf-stable.codeiumdata.com/"
+  name "Windsurf"
+  desc "Agentic IDE powered by AI Flow paradigm"
+  homepage "https://www.codeium.com/windsurf"
+
+  livecheck do
+    url "https://windsurf-stable.codeium.com/api/update/#{arch}-dmg/stable/latest"
```
Comment (khipp) [inline Casks/w/windsurf.rb:15]:
```suggestion
    url "https://windsurf-stable.codeium.com/api/update/darwin-#{arch}-dmg/stable/latest"
```

File: Casks/w/windsurf.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    strategy :json do |json|
+      "#{json["windsurfVersion"]},#{json["version"]}"
+    end
+  end
+
+  auto_updates true
+  depends_on macos: ">= :catalina"
+
+  app "Windsurf.app"
```
Comment (khipp) [inline Casks/w/windsurf.rb:24]: Since the app is a fork of VS Code, does it include a CLI?

Comment (fathiraz) [reply Casks/w/windsurf.rb:27]: Thank you for reminding me! Yes, the app includes a CLI tool! I've added the `binary` stanza to link the CLI executable:
```ruby
binary "#{appdir}/Windsurf.app/Contents/Resources/app/bin/windsurf"
```

File: Casks/w/windsurf.rb (hunk 1 above)
Comment (fathiraz) [reply Casks/w/windsurf.rb]: Thanks for the suggestion! I've updated the livecheck block to be more robust by:
- Extracting version and hash into separate variables for better readability
- Adding blank checks to ensure both values are present
- Using a cleaner string interpolation format

File: Casks/w/windsurf.rb (hunk 2 above)
Comment (fathiraz) [reply Casks/w/windsurf.rb]: Thanks for catching this! I've updated the arch identifiers to use the shorter format:
- Changed "darwin-arm64" to "arm64"
- Changed "darwin-x64" to "x64"

This indeed provides more flexibility and aligns better with common API response formats.

File: Casks/w/windsurf.rb (hunk 3 above)
Comment (fathiraz) [reply Casks/w/windsurf.rb]: Updated the URLs to include the "darwin-" prefix in both the download and livecheck URLs:
- Added "darwin-" prefix in the download URL path and filename
- Added "darwin-" prefix in the livecheck URL path
This ensures consistency with the server's expected URL format.

**Review by khipp — APPROVED** [review]
Thank you for your first contribution to `homebrew-cask`, @fathiraz! 🎉

**Pull request merged**